pkg/handlers/nutanix/mutation/machinedetails/inject.go

```diff
@@ -97,7 +97,14 @@ func (h *nutanixMachineDetailsPatchHandler) Mutate(
 			spec := obj.Spec.Template.Spec
 
 			spec.BootType = nutanixMachineDetailsVar.BootType
-			spec.Cluster = nutanixMachineDetailsVar.Cluster
+			if nutanixMachineDetailsVar.Cluster != nil {
+				spec.Cluster = *nutanixMachineDetailsVar.Cluster
+			} else {
+				// Have to set the required Type field.
+				spec.Cluster = capxv1.NutanixResourceIdentifier{
+					Type: capxv1.NutanixIdentifierName,
+				}
```

Comment:
why is this needed?

Reply:
Since the NutanixMachine.spec.Cluster is not a pointer, when it's an empty object with default field values, the Type will get the default empty string "" value, this will fail when validation with the Type field rule of the enum values. See:

type NutanixResourceIdentifier struct {
	// Type is the identifier type to use for this resource.
	// +kubebuilder:validation:Required
	// +kubebuilder:validation:Enum:=uuid;name
	Type NutanixIdentifierType `json:"type"`

Reply:
Go 1.24 can save us here https://www.bytesizego.com/blog/go-124-omitzero